### PR TITLE
Feature/38/create user marketplace service real impl

### PR DESCRIPTION
**Pull request by KaliszukTomasz**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,7 @@
 node_modules/
 build/
 **.eslintrc
+build/contracts/Migrations.json
```
Comment (lukaszkinder) [inline .gitignore]: whole build directory and everything below should be gitignored

File: src/app/user/marketplaces-user/marketplaces-user.component.ts [hunk 2]
```diff
@@ -10,12 +10,17 @@ import { MarketplaceService } from '../service/marketplace-service';
 export class MarketplacesUserComponent implements OnInit {
 
   filteredMarketplaces: MarketplaceMin[] = [];
-
+  marketplaces: MarketplaceMin[] = [];
   constructor(
     @Inject('MarketplaceService') private readonly marketplaceService: MarketplaceService,
-  ) { this.getMarketplacesMatchedToUser(); }
+  ) { this.marketplaceService.getAllMarketplaces().subscribe(
```
Comment (lukaszkinder) [inline src/app/user/marketplaces-user/marketplaces-user.component.ts]: move to ngOnInit

File: src/app/user/marketplaces-user/marketplaces-user.component.ts [hunk 3]
```diff
@@ -10,12 +10,17 @@ import { MarketplaceService } from '../service/marketplace-service';
...
+    res => this.filteredMarketplaces = res
+  ) }
 
   ngOnInit() {
+    this.marketplaceService.getAllMarketplaces().subscribe(
+      res => this.filteredMarketplaces = res
+    )
     this.getMarketplacesMatchedToUser();
```
Comment (lukaszkinder) [inline src/app/user/marketplaces-user/marketplaces-user.component.ts]: shouldn't be within a subscription?

File: src/app/user/service/marketplace-impl.service.ts [hunk 4]
```diff
@@ -1,27 +1,249 @@
...
+  currentUser: User;
+  marketplaceMinListSubject = new Subject<MarketplaceMin[]>();
+  marketplaceSubject = new Subject<Marketplace>();
+
+  constructor(
+    private userService: UserService,
+    private web3Service: Web3Service
+  ) {
+
+    this.userService.getCurrentUser().subscribe(user => this.currentUser = user);
```
Comment (lukaszkinder) [inline src/app/user/service/marketplace-impl.service.ts]: all in constructor extract to a new method

File: src/app/user/service/marketplace-impl.service.ts [hunk 5]
```diff
@@ -1,27 +1,249 @@
...
+
+    this.web3Service.artifactsToContract(auction_artifact)
+      .then((ContractAbstraction) => {
+        this.AuctionContract = ContractAbstraction;
+      });
+
+    this.updateMarkets();
+  }
+
+  private async getAllMarketsToSubject() {
```
Comment (lukaszkinder) [inline src/app/user/service/marketplace-impl.service.ts]: not needed

File: src/app/user/service/marketplace-impl.service.ts [hunk 6]
```diff
@@ -1,27 +1,249 @@
...
   }
 
   getMarketplace(marketplaceAddress): Observable<Marketplace> {
-    return;
+
+    var tempMarketplace = this.marketplacesStorage.find(
+      marketplace => marketplace.address === marketplaceAddress
+    )
+
+    return of(new Marketplace(tempMarketplace.address, tempMarketplace.name, tempMarketplace.auctions));
```
Comment (lukaszkinder) [inline src/app/user/service/marketplace-impl.service.ts]: fix it same as for marketplace admin  component

File: src/app/user/service/marketplace-impl.service.ts [hunk 7]
```diff
@@ -1,27 +1,249 @@
...
+    var instance = this.MarketPlaceContract.at(marketplaceAddress);
+    instance.addAuction(auctionName, { from: this.web3Service.getWeb3Config().eth.defaultAccount }).then((result) => {
+      console.log("Log z ADD AUCTION: " + JSON.stringify(result));
+      this.updateAuctionsInMarkets();
+    });
+    var mpIndex = this.marketplacesStorage.findIndex(
+      marketplace => marketplace.address === marketplaceAddress
+    )
+    var tempMarketplace: Marketplace = MarketplaceFactory.createMarketplace(this.marketplacesStorage[mpIndex]);
+    return of();
```
Comment (lukaszkinder) [inline src/app/user/service/marketplace-impl.service.ts]: WTF?

File: src/app/user/service/marketplace-impl.service.ts [hunk 8]
```diff
@@ -1,27 +1,249 @@
...
+                  )
+                });
+              });
+            }
+          }
+        });
+        return instance;
+      });
+
+    } catch (e) {
```
Comment (lukaszkinder) [inline src/app/user/service/marketplace-impl.service.ts]: try catch - remove

File: src/app/user/service/marketplace-impl.service.ts [hunk 9]
```diff
@@ -1,27 +1,249 @@
...
+            }
+            console.log("TERAZ" + JSON.stringify(auction));
+            console.log("log 2" + auction);
+            console.log("ilosc mp: " + marketplaceStorage.address + " " + i + this.marketplacesStorage[MPSIndex].auctions);
+
+          }
+        })
+      })
+      return 1;
+    } catch (e) {
```
Comment (lukaszkinder) [inline src/app/user/service/marketplace-impl.service.ts]: try catch ...

File: src/app/user/service/marketplace-impl.service.ts [hunk 10]
```diff
@@ -1,27 +1,249 @@
...
+              this.marketplacesStorage[MPSIndex].auctions.unshift(auction);
+            }
+            console.log("TERAZ" + JSON.stringify(auction));
+            console.log("log 2" + auction);
+            console.log("ilosc mp: " + marketplaceStorage.address + " " + i + this.marketplacesStorage[MPSIndex].auctions);
+
+          }
+        })
+      })
+      return 1;
```
Comment (lukaszkinder) [inline src/app/user/service/marketplace-impl.service.ts]: WTF?

File: src/app/user/service/marketplace-impl.service.ts [hunk 11]
```diff
@@ -1,27 +1,249 @@
...
+              if (eventResult[i].blockNumber > mapWithLastRegister[userToEraseAddress]) {
+                var userIndex = marketplaceStorage.registeredUsers.findIndex(user => user.address == userToEraseAddress);
+                marketplaceStorage.registeredUsers.splice(userIndex, 1);
+              }
+            }
+          }
+        })
+      })
+      return 1;
+    } catch (e) {
```
Comment (lukaszkinder) [inline src/app/user/service/marketplace-impl.service.ts]: TRy catch...

File: .gitignore (hunk 1 above)
Comment (KaliszukTomasz) [reply .gitignore]: ok

**Pull request merged**